packages/shared/getComponentName.js

```diff
@@ -21,36 +22,49 @@ import {
   REACT_PLACEHOLDER_TYPE,
 } from 'shared/ReactSymbols';
 
-function getComponentName(fiber: Fiber): string | null {
-  const {type} = fiber;
+function getComponentName(type: mixed): string | null {
+  if (type == null) {
+    // Host root, text node or just invalid type.
+    return null;
+  }
+  if (__DEV__) {
+    if (typeof (type: any).tag === 'number') {
+      warning(
+        false,
+        'Received an unexpected object in getComponentName(). ' +
+          'This is likely a bug in React. Please file an issue.',
+      );
+    }
+  }
   if (typeof type === 'function') {
-    return type.displayName || type.name;
+    return type.displayName || type.name || null;
   }
   if (typeof type === 'string') {
     return type;
   }
   switch (type) {
     case REACT_ASYNC_MODE_TYPE:
       return 'AsyncMode';
-    case REACT_CONTEXT_TYPE:
-      return 'Context.Consumer';
     case REACT_FRAGMENT_TYPE:
-      return 'ReactFragment';
+      return 'Fragment';
     case REACT_PORTAL_TYPE:
-      return 'ReactPortal';
+      return 'Portal';
     case REACT_PROFILER_TYPE:
-      return `Profiler(${fiber.pendingProps.id})`;
```

Comment:
If you feel strongly about keeping this, I can add an optional props argument. I'm not sure this is important. I like that having a type is sufficient (which lets us use this utility in SSR, for example).

Reply:
No, it's fine. I was just following the forward ref precedent. I think it's mildly useful, but not enough to warrant making the callsites more complicated.

Reply:
It provided a slightly more detailed entry in the component stack for Profiler components. I don't think it's very important, and I think we can drop this.